server/pulp/server/async/app.py

```diff
@@ -72,7 +166,9 @@ def initialize_worker(sender, instance, **kwargs):
     initialization.initialize()
 
     # Delete any potential old state
-    tasks._delete_worker(sender, normal_shutdown=True)
+    existing_worker = Worker.objects(name=sender).first()
+    if existing_worker:
```

Comment:
In hindsight we should have avoided moving this code around. On Pulp3 this code has been significantly reorganized already and that work is done. I only thought of this because your idea to move non task code like `_delete_worker()` and remove the _ is exactly what we did. In thinking about this more, these moves will cause merge conflicts next week when I go to merge master -> 3.0-dev which we will then have to manually resolve. I suspect we should undo the non-required moves in this PR. We could also leave this as-is, but I'll need you to resolve the merge conflicts next week so the work is unavoidable. Which of these options (or another option) would you prefer?